### PR TITLE
Add resize SDK API for storage pool

### PR DESCRIPTION
**Pull request by harsh-px**
Signed-off-by: Harsh Desai <harsh@portworx.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** (optional)
Closes #

**Special notes for your reviewer**:

### CODE REVIEW

File: cmd/osd/main.go [hunk 1]
```diff
@@ -477,16 +479,20 @@ func start(c *cli.Context) error {
 			return fmt.Errorf("Unable to Initialise Storage Policy Manager Instances %v", err)
 		}
 
+		poolServer := mockapi.NewMockOpenStoragePoolServer(gomock.NewController(
```
Comment (adityadani) [inline cmd/osd/main.go]: This should not be mock but an actual server.

File: api/api.proto [hunk 2]
```diff
@@ -232,6 +232,8 @@ message StoragePool {
   uint64 Used = 8;
   // Labels is a list of user defined name-value pairs
   map<string, string> labels = 9;
+  // UUID is the unique identifier for a storage pool
+  string UUID = 10;
```
[the remark this answers is not in the document]
Comment (harsh-px) [reply api/api.proto]: fixed

File: api/api.proto [hunk 3]
```diff
@@ -2716,6 +2731,32 @@ message SdkNodeInspectRequest {
...
+  // UUID of the storage pool to inspect
+  string uuid = 1;
+  // Size is the new desired size of the storage pool
+  uint64 size = 2;
+  // OperationType is the operation that's used to resize the storage pool (optional)
+  StoragePoolResizeOperationType operation_type = 3;
+}
+
+// Defines the operation types available to resize a storage pool
+enum StoragePoolResizeOperationType {
```
[the remark this answers is not in the document]
Comment (harsh-px) [reply api/api.proto]: done. Moved newly added enums inside messages.

File: api/server/sdk/server.go [hunk 4]
```diff
@@ -106,6 +106,8 @@ type ServerConfig struct {
 	AlertsFilterDeleter alerts.FilterDeleter
 	// StoragePolicy Manager
 	StoragePolicy policy.PolicyManager
+	// StoragePoolServer is the interface to manage storage pools in the cluster
+	StoragePoolServer api.OpenStoragePoolServer
```
[the remark this answers is not in the document]
Comment (harsh-px) [reply api/server/sdk/server.go]: updated PR now handles nil pool server handler.

File: api/server/sdk/server.go [hunk 5]
```diff
@@ -468,6 +473,7 @@ func (s *sdkGrpcServer) Start() error {
 		api.RegisterOpenStorageClusterPairServer(grpcServer, s.clusterPairServer)
 		api.RegisterOpenStoragePolicyServer(grpcServer, s.policyServer)
 		api.RegisterOpenStorageClusterDomainsServer(grpcServer, s.clusterDomainsServer)
+		api.RegisterOpenStoragePoolServer(grpcServer, s.storagePoolServer)
```
[the remark this answers is not in the document]
Comment (harsh-px) [reply api/server/sdk/server.go]: updated PR now handles nil pool server handler.

File: cmd/osd/main.go (hunk 1 above)
Comment (harsh-px) [reply cmd/osd/main.go]: removed this from osd main. nil server handler is now being handled.

**Pull request merged**